### DISCUSSION TITLE
Subdocuments while Indexing

### DISCUSSION THREAD

**Discussion by vrdn-23** [castorini/pyserini#1372] [Q&A] [answered]
Is there a way to handle or split documents into subdocuments or passages using Pyserini? I can see that during each we have an option to return the passage with max score. Could someone point me to any documentation or examples for how we deal with subdocuments or passages for a really long document?

**Comment by lintool** [comment, marked as answer]
Hi @vrdn-23 - you'll have to chop up the documents yourself into passages (i.e., create a variant corpus and index that). Number the ids like "doc1#0", "doc2#1", "doc#2"... and then you can use the max passage feature in Pyserini. See https://castorini.github.io/pyserini/2cr/msmarco-v1-doc.html for example invocations, under the "doc segmented" conditions.
